Trac3057

ostolop opened the pull request:
@alf239 please review

ostolop committed:
Fix for #3057 part 1: incorrect checker of consistency of EFV/SCV between DB and NetCDF.

ostolop committed:
Added Hibernate sessions to ExperimentAnalyticsGeneratorService. Need to cherry-pick everywhere else.

ostolop committed:
Fix for #3057 part 2: correct extraction of sample/assay properties for writing in consistent order.

ostolop committed:
Fix for #3057 part 3: incorrect checker of consistency of EFV/SCV between DB and NetCDF (cont'd).

ostolop committed:
Fix for #3057 part 4: better splitting on || and catching AtlasDataException.

ostolop committed:
long -> Long

ostolop committed:
"Final" Fix for #3057: disable analytics preservation -- too broken to be fixed easily for release.

ostolop committed:
Merge branch 'develop' into trac3057

Commit:
Made changes isolated, so that diff could be reviewed

Commit:
Code style

alf239 commented on atlas-data-storage/src/main/java/uk/ac/ebi/gxa/data/NetCDFProxy.java:
```diff
@@ -378,17 +378,20 @@ public final class NetCDFProxy implements Closeable {
+
+            if (!"".equals(data[1])) {
+                list.add(new KeyValuePair(data[0], data[1]));
             }
```
Am I right that we silently ignore all the `uVAL`s where `"".equals(data[1])`? Is it as intended?

ostolop replied:
Correct. It means that the uVAL was like "foo||" -- i.e. had an empty second value. These we ignore.

Commit:
Commented the change, removed the now-obsolete changes in  `getValuePatterns`

Commit:
Dropping the rest of not-yet-fixed code, adding references to before- and after-change revisions

 - 48f0df44ce1fbaea42dff50167827d0138bd4eb1 for an attempt to fix it
 - 05be531ebb5a93df06d6045f982d0b25e4008a11 for nearly-original version

Commit:
NetCDF updater only needs shallow ArrayDesign
